system/security/shared/caching.cpp

```diff
@@ -537,6 +518,33 @@ inline void CPermissionsCache::removeAllManagedFileScopes()
     m_managedFileScopesMap.clear();
 }
 
+static CriticalSection setCacheTimeoutCriticalSection;
+void  CPermissionsCache::setCacheTimeout(int timeoutSeconds)
+{
+    CriticalBlock setCacheTimeoutBlock(setCacheTimeoutCriticalSection);
+    m_cacheTimeoutInSeconds = timeoutSeconds;
```

Comment:
m_cacheTimeoutInSeconds should be an atomic really (should be an unsigned too really), because this may be called when the thread is running (I know it's currently only called by init, but..)